R/pkg_ref_cache_remote_checks.R

```diff
@@ -0,0 +1,45 @@
+#' Retrieve a CRAN or Bioc checks or run R CMD check
+#'
+#' @inheritParams pkg_ref_cache
+#' @family package reference cache
+#'
+pkg_ref_cache.cran_checks <- function (x, ...)
+{
+   UseMethod("pkg_ref_cache.cran_checks")
+}
+
+pkg_ref_cache.cran_checks.default <- function (x, ...) {
+  return(NA)
+}
+
+#' @importFrom xml2 read_html xml_find_all xml_text
+pkg_ref_cache.cran_checks.pkg_cran_remote <- function(x, ...) {
+  URLbase <- "https://cran.r-project.org/web/checks/check_results_"
+  webURL <- paste0(URLbase, x$name, ".html")
+
+  page <- read_html(webURL)
+  tables <- xml_find_all(page, ".//table")
+  table_cran <- xml_find_all(tables[[1]], "//tr")
+  fields <- lapply(table_cran, xml_find_all, ".//td|.//th")
+  fields <- lapply(fields, xml_text, trim=T)
```

Comment:
```suggestion
  tables <- xml2::xml_find_all(page, ".//table")
  table_cran <- xml2::xml_find_all(tables[[1]], "//tr")
  fields <- lapply(table_cran, xml2::xml_find_all, ".//td|.//th")
  fields <- lapply(fields, xml2::xml_text, trim = TRUE)
```

Reply:
Just to give some rationale, I really prefer being explicit (even though it's also itemized in the `@importFrom` section because it's still possible to have namespace conflicts. The classic example is if we imported `dplyr::filter`, but had reason to use `base::filter`. Since xml2 prefixes all of its functions with `xml_`, it's highly unlikely that we incur another dependency with namespace conflicts, but it's a good practice to get into just to avoid those rare scenarios altogether.

Reply:
Makes perfect sense, I can certainly be explicit with function calls.